CMakeLists.txt

```diff
@@ -228,6 +229,9 @@ IF (KOKKOS_HAS_TRILINOS)
   # we have to match the annoying behavior, also we have to preserve quotes
   # which needs another workaround.
   SET(KOKKOS_COMPILE_OPTIONS_TMP)
+  IF (KOKKOS_ENABLE_HIP)
+    LIST(APPEND KOKKOS_COMPILE_OPTIONS ${KOKKOS_AMDGPU_OPTIONS})
```

Comment:
Can you confirm that is the right thing to do
CUDA seems to be using `KOKKOS_ALL_COMPILE_OPTIONS`
https://github.com/kokkos/kokkos/blob/1a63570f2b74ff921e4b0dcb3f3d8e9cb4c0f05e/CMakeLists.txt#L246

Reply:
I think this has something to do with whether its ; separated list or not or so?

Reply:
I am asking because I saw https://github.com/kokkos/kokkos/blob/1a63570f2b74ff921e4b0dcb3f3d8e9cb4c0f05e/cmake/KokkosTrilinosConfig.cmake.in#L14
Should we comment then to make sure it does not look like a mistake?

Reply:
KOKKOS_COMPILER_OPTIONS is appended to KOKKOS_ALL_COMPILE_OPTIONS just a few lines below.

Reply:
OK I see https://github.com/kokkos/kokkos/blob/1a63570f2b74ff921e4b0dcb3f3d8e9cb4c0f05e/CMakeLists.txt#L244
I don't even want to look why we would need both `KOKKOS_ALL_COMPILE_OPTIONS` and `KOKKOS_COMPILER_OPTIONS `